sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOIT.java

```diff
@@ -148,6 +150,18 @@ private void gatherAndPublishMetrics(PipelineResult writeResult, PipelineResult
           return NamedTestResult.create(
               uuid, timestamp, "write_time", (writeEnd - writeStart) / 1e3);
         });
+
+    suppliers.add(
+        ignore -> {
+          try {
+            Long tableSizeAfterWrite =
+                DatabaseTestHelper.getPostgresTableSize(dataSource, tableName);
+            return NamedTestResult.create(
+                uuid, timestamp, "total_size", tableSizeAfterWrite - tableSize);
+          } catch (SQLException e) {
```

Comment:
In this case optional could be useful too. The `getPostgresTableSize` could catch the error and return `empty()` in case of problems. Then, here, you could decide if to publish the metric or not, or even fail the whole test, depending on whether you have all the necessary metrics collected.

Reply:
You're right, I changed it